madmas/nativescript-pdf-view
# NS 7.0 with Angular 10 support

Hi madmas, 

I created a PR for NativeScript 7.0 which contains updated packages, further flattened imports and an additional build targeting Angular 10 applications.

To test the local demo application:
```
git clone https://github.com/shansb-dev/nativescript-pdf-view.git
cd nativescript-pdf-view
git checkout ns-7.0
cd src
npm run clean-setup
cd demo
npm install
ns run ios
ns run android
```

To test it with an angular sampleapp make sure to first create an Angular 10 compatible bundeled version.
From nativescript-pdf-view/src:
`npm run ng-packagr`

Create the sampleapp outside the nativescript-pdf-view project:
`ns create sampleapp --ng`

Include in tsconfig.json "include" property:
`"../nativescript-pdf-view/src"`

Add dependency in package.json:
`"nativescript-pdf-view": "file:../nativescript-pdf-view/src",`

Integrate PdfView call in sampleapp:
Add pdfUrl and onLoad() to item-detail.component.ts
```
public pdfUrl = 'https://www.princexml.com/howcome/2016/samples/magic8/index.pdf';

onLoad() {
   alert('Loaded PDF!');
}
```

Add PdfView in item-detail.component.html:
`<PDFView flexgrow="1" [src]="pdfUrl" (load)="onLoad()"></PDFView>
`

Add PDFViewModule to imports in app.module.ts:
```
import { PDFViewModule } from "nativescript-pdf-view/angular/dist";

@NgModule({
    ...
    imports: [
        ...
        PDFViewModule
    ],
    ...
})
```

Then install and run from sampleapp/src/app:
```
npm install
ns run ios
ns run android
```

When running it on an ios simulator on my machine I had an issue with @nativescript/ios@7.0.1 resulting in:
```
warning: There are no architectures to compile for because all architectures in VALID_ARCHS (i386, x86_64) are also in EXCLUDED_ARCHS (i386, armv6, armv8, x86_64, arm64e). (in target 'sampleapp' from project 'sampleapp')
** BUILD FAILED **
```
This can be solved by opening sampleapp/platforms/ios/sampleapp/xcworkspace go to User-Defined Build Settings and clearing the excluded archs field.


Please let me know if further changes are required.

Edit: Would fix Issue #23
